sdk/go/common/resource/plugin/langruntime.go

```diff
@@ -16,22 +16,80 @@ package plugin
 
 import (
 	"context"
+	"fmt"
 	"io"
 
 	"github.com/blang/semver"
 	"github.com/hashicorp/hcl/v2"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	structpb "google.golang.org/protobuf/types/known/structpb"
 )
 
+// ProgramInfo contains minimal information about the program to be run.
+type ProgramInfo struct {
+	root       string
+	program    string
+	entryPoint string
+	options    map[string]any
+}
+
+func NewProgramInfo(rootDirectory, programDirectory, entryPoint string, options map[string]any) ProgramInfo {
+	return ProgramInfo{
+		root:       rootDirectory,
+		program:    programDirectory,
+		entryPoint: entryPoint,
+		options:    options,
+	}
+}
+
+// The programs root directory, i.e. where the Pulumi.yaml file is.
+func (info ProgramInfo) RootDirectory() string {
+	return info.root
+}
+
+// The programs directory, generally the same or a subdirectory of the root directory.
+func (info ProgramInfo) ProgramDirectory() string {
+	return info.program
+}
+
+// The programs main entrypoint, i.e. the path to the main program file relative to program directory.
+func (info ProgramInfo) EntryPoint() string {
+	return info.entryPoint
+}
+
+// Execution runtime options for the program.
+func (info ProgramInfo) Options() map[string]any {
+	return info.options
+}
+
+func (info ProgramInfo) String() string {
+	return fmt.Sprintf("root=%s, program=%s, entryPoint=%s", info.root, info.program, info.entryPoint)
+}
+
+func (info ProgramInfo) Marshal() (*pulumirpc.ProgramInfo, error) {
+	opts, err := structpb.NewStruct(info.options)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal options: %w", err)
+	}
+
+	return &pulumirpc.ProgramInfo{
+		RootDirectory:    info.root,
+		ProgramDirectory: info.program,
+		EntryPoint:       info.entryPoint,
+		Options:          opts,
+	}, nil
+}
+
 // LanguageRuntime is a convenient interface for interacting with language runtime plugins.  These tend to be
 // dynamically loaded as plugins, although this interface hides this fact from the calling code.
 type LanguageRuntime interface {
 	// Closer closes any underlying OS resources associated with this plugin (like processes, RPC channels, etc).
 	io.Closer
 	// GetRequiredPlugins computes the complete set of anticipated plugins required by a program.
-	GetRequiredPlugins(info ProgInfo) ([]workspace.PluginSpec, error)
+	GetRequiredPlugins(project string, info ProgramInfo) ([]workspace.PluginSpec, error)
```

Comment:
No need to pass project here, I've deprecated it so the engine just hardcodes it to "deprecated".

Reply:
Done ✅